v1/ao/internal/config/config.go

```diff
@@ -374,9 +374,13 @@ func (c *Config) validate() error {
 	}
 
 	if c.ReporterType != reporterTypeServerless {
-		c.ServiceKey = ToServiceKey(c.ServiceKey)
-		if ok := IsValidServiceKey(c.ServiceKey); !ok {
-			return errors.Wrap(ErrInvalidServiceKey, fmt.Sprintf("service key: \"%s\"", c.ServiceKey))
+		if c.ServiceKey != "" {
+			c.ServiceKey = ToServiceKey(c.ServiceKey)
+			if ok := IsValidServiceKey(c.ServiceKey); !ok {
+				return errors.Wrap(ErrInvalidServiceKey, fmt.Sprintf("service key: \"%s\"", c.ServiceKey))
+			}
+		} else {
+			log.Warning("Service key is not assigned.")
```

Comment:
Just curious whether this "set service key later" is an option provided to user? If so empty service key from config is actually okay (therefore we might consider something less severe)?

Reply:
It makes sense. I'll change it to Info.